pyro/infer/mcmc/mcmc.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import, division, print_function
+
+import logging
+
+from pyro.infer import TracePosterior
+from pyro import poutine
+
+
+class MCMC(TracePosterior):
+    """
+    Wrapper class for Markov Chain Monte Carlo algorithms. Specific MCMC algorithms
+    are TraceKernel instances and need to be supplied as a ``kernel`` argument
+    to the constructor.
+
+    :param model: A stochastic function containing Pyro primitive calls.
+    :param kernel: An implementation of the ``TraceKernel`` class, which when
+        given an execution trace returns another sample trace from the target
+        (posterior) distribution.
+    :param int num_samples: The number of samples that need to be generated,
+        excluding the samples discarded during the warmup phase.
+    :param int warmup_steps: Number of warmup iterations. The samples generated
+        during the warmup phase are discarded.
+    :param \*args: Positional arguments for constructing the ``kernel`` instance.
+    :param \*\*kwargs: Arbitrary keyword arguments for constructing the ``kernel``
+        instance.
+    """
+
+    def __init__(self, model, kernel, warmup_steps, num_samples, *args, **kwargs):
```

Comment:
I would recommend putting `num_samples` first, defaulting `warmum_steps=0`. (In the future we might also add a `take_every=1` argument to allow taking only every few samples.) I would also recommend passing in a constructed kernel rather than a cumbersome kernel factory + `*args, **kwargs` (which will inevitably lead to headaches as we change the `MCMC` interface).

Reply:
Makes sense, will address.